pkg/health/health.go

```diff
@@ -26,6 +28,10 @@ const (
 	HealthStatusMissing HealthStatusCode = "Missing"
 )
 
+// enableArgoCronWorkflowHealthCheck is an environment variable to control whether to enable health check for
+// Argo CronWorkflow. Defaults to false.
+var enableArgoCronWorkflowHealthCheck = os.Getenv("ENABLE_ARGO_CRON_WORKFLOW_HEALTH_CHECK") == "true"
```

Comment:
@jessesuen Is this what you meant? Should the name be more generic so we can reuse it later for resources like `CronJob`?